### PR TITLE
Weapon Select - Use `selectThrowable` command

### PR DESCRIPTION
**Pull request by LinkIsGrim**
**When merged this pull request will:**
- Title.

~~Currently bugged with IR grenades (seems they need to be a specific side?), see: https://feedback.bistudio.com/T176248~~ Fixed by BI in rev. 151086

### IMPORTANT

- [ ] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [x] [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/weaponselect/functions/fnc_selectNextGrenade.sqf [hunk 1]
```diff
@@ -28,17 +28,10 @@ if (_currentGrenade isEqualTo []) then {
...
-private _grenades = [];
-
-{
-    if (_x in _magazines) then {
-        _grenades pushBack _x;
-    };
-    false
-} count ([GVAR(GrenadesAll), GVAR(GrenadesFrag), GVAR(GrenadesNonFrag)] select _type);
+// This is faster than checking magazines
+private _grenades = (_unit weaponsInfo ["Throw", true]) select {(_x select 5) in _grenadeType} apply {_x select 5};
```
Comment (johnb432) [inline addons/weaponselect/functions/fnc_selectNextGrenade.sqf:34]:
```suggestion
private _grenades = (_unit weaponsInfo ["Throw", true]) apply {_x select 5} select {_x in _grenadeType} ;
```

**Comment by LinkIsGrim** [comment]
From looking at feedback tracker, seems like using muzzles directly might work instead? Would be better if it was fixed anyway.

File: addons/weaponselect/functions/fnc_selectNextGrenade.sqf [hunk 2]
```diff
@@ -52,24 +46,6 @@ if (_nextGrenadeIndex >= count _grenades) then {
...
-{_unit removeItemFromVest     _x} forEach _vestGrenades;
-{_unit removeItemFromBackpack _x} forEach _backpackGrenades;
-
-// readd grenades
-{_unit addItemToUniform  _x} forEach _uniformGrenades;
-{_unit addItemToVest     _x} forEach _vestGrenades;
-{_unit addItemToBackpack _x} forEach _backpackGrenades;
-
-[_nextGrenade, {_x == _nextGrenade} count _magazines] call FUNC(displayGrenadeTypeAndNumber);
+[_nextGrenade, {_x == _nextGrenade} count (magazines _unit)] call FUNC(displayGrenadeTypeAndNumber);
```
[the remark this answers is not in the document]
Comment (johnb432) [reply addons/weaponselect/functions/fnc_selectNextGrenade.sqf:49]: Not worth it imo.

**Review by johnb432 — APPROVED** [review]

**Comment by PabstMirror** [comment]
Noting that this fixes a current bug with the func in that it won't skip past chemlights
```
[["arifle_MX_ACO_pointer_F","","acc_pointer_IR","optic_Aco",["30Rnd_65x39_caseless_mag",30],[],""],[],["hgun_P07_F","","","",["16Rnd_9x21_Mag",17],[],""],["U_B_CombatUniform_mcam",[["ACE_fieldDressing",1],["ACE_packingBandage",1],["ACE_morphine",1],["ACE_tourniquet",1],["30Rnd_65x39_caseless_mag",2,30],["HandGrenade",1,1]]],["V_PlateCarrierSpec_rgr",[["30Rnd_65x39_caseless_mag",7,30],["16Rnd_9x21_Mag",2,17],["SmokeShell",1,1],["SmokeShellGreen",1,1],["Chemlight_green",2,1]]],[],"H_HelmetB","",[],["ItemMap","B_UavTerminal","ItemRadio","ItemCompass","ItemWatch","NVGoggles"]]
```

repeated calls to `[player] call ace_weaponselect_fnc_selectNextGrenade` and it will get stuck on the green chem lights

**Pull request merged**